synapse/storage/util/id_generators.py

```diff
@@ -101,7 +100,10 @@ def __init__(self, db_conn, table, column, extra_tables=[], step=1):
             self._current = (max if step > 0 else min)(
                 self._current, _load_current_id(db_conn, table, column, step)
             )
-        self._unfinished_ids = deque()  # type: Deque[int]
+
+        # We use this as an ordered set, as we want to efficiently append items,
+        # remove items and get the first item.
```

Comment:
can you explain what the values of the dict are? same as the key? `0`? the former I think?